src/script/conversation/ConversationMapper.ts

```diff
@@ -489,4 +469,48 @@ export class ConversationMapper {
       : ACCESS_STATE.PERSONAL.ONE2ONE;
     return conversationEntity.accessState(personalAccessState);
   }
+
+  private static mapAccessStateV2(conversationEntity: Conversation, accessRole: ACCESS_ROLE_V2[]) {
```

Comment:
Maybe those 2 functions `mapAccessStaetV2` and `mapLegacyAccessState` could be pure and just return an `ACCESS_STATE`.

Then we would have a single call to `conversation.accessState(newAccessState)` in the parent function? 
You would also not need to inject `conversationEntity` into those functions. WDYT?

Reply:
Makes sense :D